packages/headless/src/features/facets/range-facets/date-facet-set/date-facet-set-slice.ts

```diff
@@ -40,14 +41,30 @@ export const dateFacetSetReducer = createReducer(
       )
       .addCase(restoreSearchParameters, (state, action) => {
         const df = action.payload.df || {};
-        handleRangeFacetSearchParameterRestoration(state, df);
+        const facetIds = Object.keys(state);
+
+        facetIds.forEach((id) => {
+          const request = state[id];
+          const paramsValues = df[id] || [];
+
+          request.currentValues = handleRangeFacetSearchParameterRestoration(
```

Comment:
Limited the scope of the `handleRangeFacetSearchParameterRestoration` so we could provide a different compare/find method. Keeping the original scope and trying to pass a find method caused issues with the type which I was not able to resolve.